openedx/features/course_experience/course_tools.py

```diff
@@ -0,0 +1,52 @@
+"""
+Support for course tool plugins.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+
+# Stevedore extension point namespace
+COURSE_TOOLS_NAMESPACE = 'openedx.course_tool'
+
+
+class CourseTool(object):
+    """
+    This is an optional base class for Course Tool plugins.
+
+    Plugin implementations can choose to subclass CourseTool to get
+    useful default behavior, but it is not a requirement. A course tool
+    plugin should define the following class methods:
+
+     * is_enabled(cls, course_key):
+       Returns true if this tool is enabled for the specified course key.
+     * title(cls):
+       Returns the title of this tool.
+     * icon_classes(cls):
+       Returns the icon classes needed to represent this tool.
+     * url(cls, course_key):
+       Returns the URL for this tool for the specified course key.
+    """
```

Comment:
I was wondering where this convention came from. I'd never seen new lines at the end of docstrings but noticed recently that you'd started doing it. It isn't part of the PEP8 standard:

https://www.python.org/dev/peps/pep-0257/

We base our style on Google's which also doesn't have an extra newline:

https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

It just looks wrong to me, but I do see that it is in our documentation:

https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#docstrings